queries/highlights.scm

```diff
@@ -1,76 +1,104 @@
-; Comments
-(line_comment) @comment
+(comment) @comment
 
-(block_comment) @comment
+(ty (id)) @type
+
+(valid_semver) @property
+
+(world_item
+  name: (id) @property)
+
+(interface_item
+  name: (id) @property)
+
+(type_item
+  alias: (id) @property)
+
+(func_item
+  name: (id) @function)
+
+(handle
+  (id) @type)
+
+(named_type
+  name: (id) @variable.parameter)
+
+(record_item
+  name: (id) @variable)
+
+(record_field
+  name: (id) @variable)
+
+(flags_items
+  name: (id) @variable)
+
+(flags_body
+  (id) @property)
+
+(variant_items
+  name: (id) @variable)
+
+(variant_case
+    name: (id) @variable)
+
+(enum_items
+  name: (id) @variable)
+
+(enum_body
+  enum_cases: (id) @property)
+
+(resource_method
+  "constructor" @constructor)
+
+[
+  "type"
+  "resource"
+  "func"
+  "record"
+  "enum"
+  "flags"
+  "variant"
+  "static"
+  "interface"
+  "world"
+  "import"
+  "export"
+  "package"
+  "include"
+] @keyword
 
-; Primitive Types
 [
-  "bool"
-  "s8"
-  "s16"
-  "s32"
-  "s64"
   "u8"
   "u16"
   "u32"
   "u64"
-  "float32"
-  "float64"
+  "s8"
+  "s16"
+  "s32"
+  "s64"
+  "f32"
+  "f64"
   "char"
+  "bool"
   "string"
-  ; Container Types
-  "list"
   "tuple"
+  "list"
   "option"
   "result"
-] @type.builtin
-
-"func" @keyword.function
-
-; Keywords for file structure and components
-[
-  "record"
-  "enum"
-  "variant"
-  "flags"
-  "resource"
-] @keyword.type
-
-; Keywords for importing and exporting
-[
-  "package"
-  "world"
-  "use"
-  "import"
-] @keyword.import
-
-; Resource Keywords
-"static" @keyword.modifier
-
-; Named Types (Capitalized identifiers)
-((identifier) @type
-  (#match? @type "^[A-Z]"))
-
-((identifier) @variable
-  (#match? @variable "^[a-z_][a-zA-Z0-9_]*$"))
-
-; Constants (UPPER_CASE names and Enums)
-((identifier) @constant
-  (#match? @constant "^[A-Z][A-Z0-9_]+$"))
-
-; Functions and Methods (lowercase names followed by parentheses)
-((identifier) @function
-  (#match? @function "^[a-z_][a-zA-Z0-9_]*%("))
+  "borrow"
+] @type
```

Comment:
Please ensure you conform to [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md) standards, so this should be @type.builtin

Reply:
I was a bit lazy on highlighting but tried to do it better this time.